lib/cache/cache.go

```diff
@@ -235,6 +235,11 @@ func ForNode(cfg Config) Config {
 		{Kind: types.KindNamespace, Name: apidefaults.Namespace},
 		{Kind: types.KindNetworkRestrictions},
 	}
+
+	if cfg.DiscoveryAgent {
```

Comment:
They can deploy one or multiple, for example each discovering a subset of the nodes.

Yes, I did consider making it a separate service when working on database discovery but it felt like having a database agent run discovery for the databases it will proxy makes sense so we took the same approach for SSH. I think it's ok for now, but there's definitely merit to this idea, we'll probably revisit in the course of working on Teleport Discover.

Reply:
Yes, exactly - the reason agents run discovery is to support Cloud. Still in theory would make sense to have `discovery_service` in agents which would support discovery for all types of resources we have, that would be quite significant refactoring though which is why I think we should do it when working on "day 2" Teleport Discover initiative (which will focus on auto-discovery).

Reply:
We want to get this in 10.2 so it's a little too late for a redesign like this IMO.